src/docs/arch-schema.dox

```diff
@@ -1,11 +1,91 @@
+ - Row-oriented table keeps all the data in one file called \c "<table name>.wt", where
+ \c "<table name>" is the name that was passed as a part of \c name parameter to WT_SESSION::create.
+ - Column-oriented table stores the data in multiple files. One for each column group. Filename
+ will be as follows \c "<table name>_<colgroup name>.wt". Where \c "<table name>" is the name that
+ was specified as a part of \c name parameter to WT_SESSION::create. And \c "<colgroup name>" is the
+ column group name defined in \c colgroups entry during the definition of the table format.
+ See this example of how column groups can be configured in WiredTiger @ex_ref{ex_col_store.c}.
+
+@ref arch-row-column describes in more detail how row and column stores work.
+
+@section schema_integrity Schema Integrity
+
+A user can create and manipulate database objects through the API listed on this page. There are
+several WiredTiger internal objects such as @ref arch-metadata, @ref arch-hs, etc. The schema of
+those objects is locked and cannot be altered from outside of WiredTiger.
+
+Schema operations cause an update to the metadata and are performed under a schema lock to avoid
+concurrent operations on the same object. The following sequence of steps define a generic schema
+operation:
+
+@plantuml_start{schema_generic.png }
+@startuml{schema_generic.png}
+:A schema operation;
+partition with-schema-lock {
+    :perform operation on underlying data-object;
+    :update metadata-file;
+    :checkpoint and sync metadata;
+}
+stop
+@enduml
+@plantuml_end
+
+Apart from the schema API, a schema lock is necessary for many other
+operations in WiredTiger including the following "heavy" database modifications:
+ - A global schema lock wraps \c checkpoint \c prepare to avoid any tables being created or dropped
+ during this phase. See @ref arch-checkpoint for details.
+ - \c Rollback \c to \c stable operation acquires a global schema lock to make sure no schema changes are
+ done during this complex process. @ref arch-rts has more information on the operation.
+ - A \c backup \c cursor, when opened, ensures that it is the only backup cursor running in the
+ system. To achieve that it holds a schema lock. See @ref arch_cursor_backup for more
+ information on backup cursors.
+
+All the schema operations listed below perform multi-step metadata modifications. However, they are
+\c non-transactional and will not guarantee ACID properties, see @ref transactions for more
+details. This means that if the application stops abnormally during schema manipulations,
+metadata may be in an invalid state. WiredTiger may or may not be able to recover after this
+type of crash.
+
+@section schema_api Schema Manipulation
+
+All schema manipulations are done in the context of WT_SESSION.
+
+@subsection schema_create Create
+
+The create schema operation is responsible for creating the underlying data
+object on the filesystem with the right parameters and then creating an entry
```

Comment:
"the underlying data object..." implies a create call always creates only 1 single object in the file system and that may not be the case. Also a single create call for the simplest row-store table creates 3 entries in the metadata file. So please reword this paragraph to account for that.